src/WaCore.Crud.Services/WacListDataService.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using WaCore.Crud.Contracts.Data;
+using WaCore.Crud.Contracts.Services;
+
+namespace WaCore.Crud.Services
+{
+    public abstract class WacListDataService<TEntity, TFilter, TDto> : IWacListDataService<TFilter, TDto>
+    {
+        private readonly IWacListDataRepository<TEntity, TFilter> repo;
+
+        public WacListDataService(IWacListDataRepository<TEntity, TFilter> repo)
+        {
+            this.repo = repo;
```

Comment:
In case of leading underscores it's not necessary to use 'this' keyword. We can do like this: _repo = repo;

Reply:
again same as above...